web/components/contract/contract-card.tsx

```diff
@@ -378,80 +382,78 @@ export function ContractCardProbChange(props: {
   showImage?: boolean
 }) {
   const { noLinkAvatar, showPosition, className, showImage } = props
+  const contract = (useContract(props.contract.id) ??
+    props.contract) as CPMMBinaryContract
+
   const yesOutcomeLabel =
     props.contract.outcomeType === 'PSEUDO_NUMERIC' ? 'HIGHER' : 'YES'
   const noOutcomeLabel =
     props.contract.outcomeType === 'PSEUDO_NUMERIC' ? 'LOWER' : 'NO'
 
-  const contract = (useContract(props.contract.id) ??
-    props.contract) as CPMMBinaryContract
-
   const user = useUser()
-  const metrics = useUserContractMetrics(user?.id, contract.id)
-  const dayMetrics = metrics && metrics.from && metrics.from.day
-  const binaryOutcome =
-    metrics && floatingEqual(metrics.totalShares.NO ?? 0, 0) ? 'YES' : 'NO'
-
-  const displayedProfit = dayMetrics
-    ? ENV_CONFIG.moneyMoniker + dayMetrics.profit.toFixed(0)
-    : undefined
+  const userBets = useUserContractBets(user?.id, contract.id)
+  let yesWinnings,
+    noWinnings,
+    position,
+    outcome,
+    profit = null
+  const bets = userBets?.filter((b) => !b.isAnte)
+  if (bets) {
+    const metrics = getContractBetMetrics(contract, bets)
+    profit = metrics.profit
+    const excludeSales = bets.filter((b) => !b.isSold && !b.sale)
+    yesWinnings = sumBy(excludeSales, (bet) =>
+      calculatePayout(contract, bet, 'YES')
+    )
+    noWinnings = sumBy(excludeSales, (bet) =>
+      calculatePayout(contract, bet, 'NO')
+    )
+    position = yesWinnings - noWinnings
+    outcome = position < 0 ? 'NO' : 'YES'
+  }
```

Comment:
I would also recommend `useSaveBinaryShares` here.

The other benefit of this function is the data is cached to local storage, which makes for a much better loading experience.